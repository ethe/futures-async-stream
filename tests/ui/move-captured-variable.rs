// SPDX-License-Identifier: Apache-2.0 OR MIT

#![feature(generators)]

use futures_async_stream::stream_block;

fn foo<F: FnMut()>(_f: F) {}

fn main() {
    let a = String::new();
    foo(|| {
        stream_block! { //~ ERROR cannot move out of `a`, a captured variable in an `FnMut` closure
            yield a
        };
    });
}
