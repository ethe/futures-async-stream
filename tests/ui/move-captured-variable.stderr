error[E0507]: cannot move out of `a`, a captured variable in an `FnMut` closure
  --> tests/ui/move-captured-variable.rs:10:9
   |
8  |        let a = String::new();
   |            - captured outer variable
9  |        foo(|| {
   |   _________-
10 |  |         stream_block! { //~ ERROR cannot move out of `a`, a captured variable in an `FnMut` closure
   |  |_________^
11 | ||             yield a
   | ||                   -
   | ||                   |
   | ||                   move occurs because `a` has type `String`, which does not implement the `Copy` trait
   | ||                   move occurs due to use in generator
12 | ||         };
   | ||__________^ move out of `a` occurs here
13 |  |     });
   |  |_____- captured by this `FnMut` closure
   |
   = note: this error originates in the macro `stream_block` (in Nightly builds, run with -Z macro-backtrace for more info)
